### PR TITLE
Visit PDF Changes

### PR DESCRIPTION
**Pull request by Johngtka**
### Features List:

- [x] `changed method for displaying selected points images in pdf by filtering images for code 'P'`

### CODE REVIEW

File: src/app/visit/visit.component.ts [hunk 1]
```diff
@@ -290,9 +290,9 @@ export class VisitComponent implements OnInit, AfterViewInit, OnDestroy {
                 {
                     table: {
                         widths: ['50%', '50%'],
-                        body: this.visitPoints.map((point) =>
-                            this.getPdfRow(point._id),
-                        ),
+                        body: this.visitPoints
+                            .filter((vt) => vt.code.indexOf('P', 0))
```
Comment (macrog) [inline src/app/visit/visit.component.ts:294]: image

this is not working correctly, what does the indexOf returns?

**Review by macrog — CHANGES_REQUESTED** [review]

File: src/app/visit/visit.component.ts (hunk 1 above)
Comment (Johngtka) [reply src/app/visit/visit.component.ts:294]: index of is returns a number

Comment (macrog) [reply src/app/visit/visit.component.ts:294]: so? what number returns when character is not found? you filter logic should check if the 'P' is found or not? Current filter removes all objects that have a code starting with 'P'. When you do changes please check if it works first, set a debugger and analyse the code if it works correctly.

Comment (macrog) [reply src/app/visit/visit.component.ts:294]: any progress??

Comment (macrog) [reply src/app/visit/visit.component.ts:294]: any update?

Comment (Johngtka) [reply src/app/visit/visit.component.ts:294]: On this moment no, but i prepare to recheck all of changes in this pull request in a few days

**Comment by Johngtka** [comment]
s&m commit proposal:
`refactor: new filter in pdf image data displayed by code = 'P' (#59)`

**Review by macrog — APPROVED** [review]
👍

**Pull request merged**